src/hb-ot-cmap-table.hh

```diff
@@ -1468,7 +1503,155 @@ struct cmap
     | hb_filter ([&] (const hb_pair_t<hb_codepoint_t, hb_codepoint_t> _)
 		 { return (_.second != HB_MAP_VALUE_INVALID); })
     ;
-    cmap_prime->serialize (c->serializer, it, encodingrec_iter, this, c->plan);
+
+    for (auto _ : format12_tables.iter ())
+    {
+      unsigned language =_.second;
+      const EncodingRecord *format4_tbl = nullptr;
+      const EncodingRecord *format12_tbl = hb_addressof (encodingRecord[_.first]);
+      if (format12_tbl->platformID == 0 && format12_tbl->encodingID == 4)
+      {
+        if (!unicode_bmp_tables.has (language))
+        {
+          format12_tables.del (_.first);
```

Comment:
I don't think the table should be removed in this case. In the fontTools PR the table is only removed if there is a matching language cmap4 table.

Reply:
Actually format12_tables keeps format12 tables that should be removed. I'll rename it so it's not misleading.